runtime/j9vm31/jnicgen.cpp

```diff
@@ -0,0 +1,1520 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "j9vm31.h"
+
+static jobject
+CallObjectMethodV_64(JNIEnv *env, jobject obj, jmethodID methodID, va_list_64 parms)
+{
+	const jint NUM_ARGS = 4;
+	J9_CEL4RO64_ArgType argTypes[NUM_ARGS] = { CEL4RO64_type_JNIEnv64, CEL4RO64_type_jobject, CEL4RO64_type_jmethodID, CEL4RO64_type_uint32_ptr };
+	uint64_t argValues[NUM_ARGS] = { JNIENV64_FROM_JNIENV31(env), obj, methodID, (uint64_t)parms };
+	jobject returnValue = NULL;
+	FUNCTION_DESCRIPTOR_FROM_JNIENV31(env, CallObjectMethodV);
+	j9_cel4ro64_call_function(functionDescriptor, argTypes, argValues, NUM_ARGS, CEL4RO64_type_jobject, &returnValue);
+	return returnValue;
+}
+
+jobject JNICALL
+CallObjectMethod(JNIEnv *env, jobject obj, jmethodID methodID, ...)
+{
+	va_list_64 parms;
+	long long value = *((int *)((char*)&(methodID) + sizeof(jmethodID)));
+	(void)( (parms)[0] = 0, (parms)[1] =  (char *)&(methodID), (parms)[2] = 0, (parms)[3] = (parms)[1] + sizeof(jmethodID) );
+
+	jobject returnValue = CallObjectMethodV_64(env, obj, methodID, parms);
+	(void)( (parms)[0] = 0, (parms)[1] =  (char *)0, (parms)[2] = 0, (parms)[3] = (char *)0);
```

Comment:
What's this for?

Reply:
The last line is not necessary.  I had expanded the va_list macro and updated to 64-bit representation, but at this stage after the call, parms is no longer needed.

Reply:
Cleaned up under: 6ae3a89